lib/api.js

```diff
@@ -111,7 +111,7 @@ class JSDOM {
   static fromURL(url, options = {}) {
     return Promise.resolve().then(() => {
       const parsedURL = new URL(url);
-      url = parsedURL.href;
+      url = parsedURL.origin + parsedURL.pathname + parsedURL.search;
```

Comment:
This doesn't seem like quite the right fix; it'd be better to strip out the hash by setting it to null. I'll push a tweak that does that.

Thanks very much for the tests though; they're the key!